src/theory/theory_engine.cpp

```diff
@@ -2268,14 +2268,26 @@ void TheoryEngine::checkTheoryAssertionsWithModel(bool hardFailure) {
         Node val = getModel()->getValue(assertion);
         if (val != d_true)
         {
+          std::stringstream ss;
+          ss << theoryId
+             << " has an asserted fact that the model doesn't satisfy." << endl
+             << "The fact: " << assertion << endl
+             << "Model value: " << val << endl;
           if (hardFailure)
```

Comment:
Why do we only check this when `hardFailure` is enabled?

Reply:
I think hardFailure is impacted by whether the user asked us to check model explicitly or whether this is an internal check.